index.ts

```diff
@@ -221,18 +221,31 @@ class Applesign {
     await tools.asyncRimraf(placeholderdir);
   }
 
-  // XXX some directory leftovers
   async removeXCTests() {
     fchk(arguments, []);
+    let dirsToCheck: Set<string> = new Set<string>();
     const dir = this.config.appdir;
     walk.walkSync(dir, (basedir: string, filename: string, stat: any) => {
       const target = path.join(basedir, filename);
       //  if (target.toLowerCase().indexOf('/xct') !== -1)
       if (target.toLowerCase().indexOf("xctest") !== -1) {
         this.emit("message", "Deleting " + target);
         fs.unlinkSync(target);
+        dirsToCheck.add(basedir);
       }
     });
+    let removedDirs: Set<string> = new Set<string>();
```

Comment:
It’s unclear to me the need to define a set of paths because we can process them in the first loop already in place, and an array should be fine and simpler because you cant have two different paths with the same path